aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+        recipient: address,
+        amount: u64,
+        nonce: u64,
+    }
+
+    /// This struct will store the event handles for bridge events.
+    struct BridgeEvents has key, store {
+        bridge_transfer_initiated_events: EventHandle<BridgeTransferInitiatedEvent>,
+        bridge_transfer_completed_events: EventHandle<BridgeTransferCompletedEvent>,
+    }
+
+    struct Nonce has key {
+        value: u64
+    }
+
+    /// Increment and get the current nonce  
+    fun increment_and_get_nonce(): u64 acquires Nonce {  
+        let nonce_ref = borrow_global_mut<Nonce>(@aptos_framework);  
+        nonce_ref.value = nonce_ref.value + 1;  
+        nonce_ref.value  
+    }  
+
+    /// Initializes the module and stores the `EventHandle`s in the resource.
+    public fun initialize(aptos_framework: &signer) {
+        system_addresses::assert_aptos_framework(aptos_framework);
+
+        // Ensure the nonce is not already initialized
+        assert!(
+            !exists<Nonce>(signer::address_of(aptos_framework)),
+            2
+        );
+
+        // Create the Nonce resource with an initial value of 1
+        move_to<Nonce>(aptos_framework, Nonce { 
+            value: 1 
+        });
+
+        move_to(aptos_framework, BridgeEvents {
+            bridge_transfer_initiated_events: account::new_event_handle<BridgeTransferInitiatedEvent>(aptos_framework),
+            bridge_transfer_completed_events: account::new_event_handle<BridgeTransferCompletedEvent>(aptos_framework),
+        });
+    }
+
+    /// Initiate a bridge transfer of MOVE from Movement to the base layer  
+    /// Anyone can initiate a bridge transfer from the source chain  
+    /// The amount is burnt from the initiator and the module-level nonce is incremented  
+    /// @param initiator The initiator's Ethereum address as a vector of bytes.  
+    /// @param recipient The address of the recipient on the Aptos blockchain.  
+    /// @param amount The amount of assets to be locked.  
+    public entry fun initiate_bridge_transfer(  
+        initiator: &signer,  
+        recipient: vector<u8>,  
+        amount: u64  
+    ) acquires BridgeEvents, Nonce {  
+        let initiator_address = signer::address_of(initiator);  
+        let recipient_address = ethereum::ethereum_address_no_eip55(recipient);  
+    
+        // Increment and retrieve the nonce  
+        let nonce = increment_and_get_nonce();  
+    
+        // Create bridge transfer details  
+        let details = native_bridge_store::create_details(  
+            initiator_address,  
+            recipient_address, 
+            amount,  
+            nonce  
+        );  
+
+        let bridge_transfer_id = native_bridge_store::bridge_transfer_id(
+            initiator_address, 
+            recipient_address, 
+            amount, 
+            nonce
+        ); 
+    
+        // Add the transfer details to storage  
+        native_bridge_store::add(nonce, details);
```

Comment:
Why do we need to store the bridge details?
If there is not refund, we will never use these details in the future, won't we?